src/core/plugin/luapi_application.h

```diff
@@ -327,6 +378,335 @@ static int applib_layerAction(lua_State* L) {
+            filled = toolHandler->getHighlighterFillEnabled();
+
+            Tool& tool = toolHandler->getTool(TOOL_HIGHLIGHTER);
+            color = tool.getColor();
+            std::string drawingType = drawingTypeToString(tool.getDrawingType());
+        } else { // Tool type must be pen
+            if (!(strcmp("pen", tool) == 0))
+                g_warning("%s", FC(_F("Unknown stroke type: \"{1}\", defaulting to pen") % tool));
+
+            stroke->setToolType(STROKE_TOOL_PEN);
+
+            size         = toolSizeToString(toolHandler->getPenSize());
+            thickness    = toolHandler->getToolThickness(TOOL_PEN)[toolSizeFromString(size)];
+
+            fillOpacity  = toolHandler->getPenFill();
+            filled       = toolHandler->getPenFillEnabled();
+
+            Tool& tool   = toolHandler->getTool(TOOL_PEN);
+            color        = tool.getColor();
+            drawingType  = drawingTypeToString(tool.getDrawingType());
+            lineStyle    = StrokeStyle::formatStyle(tool.getLineStyle());
+        }
+
+        // Set width
+        if (lua_isnumber(L, -4)) // Check if the width was provided
+            stroke->setWidth(lua_tonumber(L, -4));
+        else
+            stroke->setWidth(thickness);
+
+        // Set color
+        if (lua_isinteger(L, -3)) // Check if the color was provided
+            stroke->setColor(Color(lua_tointeger(L, -3)));
+        else
+            stroke->setColor(color);
+
+        // Set fill
+        if (lua_isinteger(L, -2)) // Check if fill settings were provided
+            stroke->setFill(lua_tointeger(L, -2));
+        else if (filled)
+            stroke->setFill(fillOpacity);
+        else
+            stroke->setFill(-1); // No fill
+
+        // Set line style
+        if (lua_isstring(L, -1)) // Check if line style settings were provided
+            stroke->setLineStyle(StrokeStyle::parseStyle(lua_tostring(L, -1)));
+        else
+            stroke->setLineStyle(StrokeStyle::parseStyle(lineStyle.data()));
+
+        lua_pop(L, 5); // Finally done with all that Lua data.
+
+        // Add the Stroke
+        layer->addElement(stroke);
+        stroke = nullptr;
+        return;
+    }
+    // If there aren't at least 2 points, then don't add the stroke.
+    g_warning("Stroke shorter than two points. Discarding. (Has %d)", stroke->getPointCount());
+    delete stroke;
+    stroke = nullptr;
+    return;
+}
+
+/**
+ * Given a table containing a series of splines, draws a stroke on the canvas.
+ * Expects a table of coordinate pairs along with attributes of the stroke.
+ * Attributes are optional.
+ *
+ * The function expects 8 points per spline. Due to the nature of quadratic
+ * splines, you must pass your points in a repeating pattern:
+ * startX, startY, ctrl1X, ctrl1Y, ctrl2X, ctrl2Y, endX, endY, startX, startY, ...
+ *
+ * The function checks that the spline table is divisible by eight, and will throw
+ * an error if it is not.
+ *
```

Comment:
Please add a comment on optional values and defaults.

Reply:
I wanted you to comment, that the defaults come from the current settings of the pen/highlighter tool, if none are specified.
It should also be added, that only "pen" or "highlighter" are allowed for the tool key and that "pen" is default.